core/src/components/cat-checkbox/cat-checkbox.tsx

```diff
@@ -121,13 +123,20 @@ export class CatCheckbox {
             <polyline points="1.5 5 10.5 5"></polyline>
           </svg>
         </span>
-        <span class="label" part="label">
-          {this.label}
+        <span class={{ label: true, 'label-group': Boolean(this.label) && this.hasSlottedLabel() }} part="label">
+          {[this.label, this.hasSlottedLabel() && <slot name="label"></slot>]}
         </span>
       </label>
     );
   }
 
+  private hasSlottedLabel() {
+    return (
+      this.hostElement.children &&
```

Comment:
where did you remove it? And I think so, because the slot element is always written even when not used in the light dom and for the condition of `log.error('[A11y] Missing ARIA label on checkbox', this);`. Although I could simplify it into a variable instead of a method, like so: `hasSlottedLabel = !!this.hostElement.querySelector('[slot="label"]')`.If not, I'll be giving it a spin, maybe with the change of the exclusion I can think of something cleaner